qupath-core/src/main/java/qupath/lib/objects/utils/MeasurementStrategy.java

```diff
@@ -0,0 +1,144 @@
+
+    /**
+     * Assign the measurements of the first object to the output object.
+     */
+    MeasurementStrategy USE_FIRST = ((pathObjects, outputList) -> {
+        outputList.putAll(pathObjects.getFirst().getMeasurementList());
+    });
+
+    /**
+     * Calculate the mean of each measurement and pass these to the output object.
+     */
+    MeasurementStrategy MEAN = ((pathObjects, outputList) -> {
+        for (var name: pathObjects.getFirst().getMeasurementList().getNames()) {
+            outputList.put(
+                    name,
+                    pathObjects.stream().mapToDouble(po -> po.getMeasurementList().get(name)).average().orElse(Double.NaN)
+            );
+        }
+    });
+
+    /**
+     * Calculate a mean weighted by object area and pass this to the output object.
+     */
+    MeasurementStrategy WEIGHTED_MEAN = ((pathObjects, outputList) -> {
+        var weights = computeAreaWeights(pathObjects);
+        for (var name: pathObjects.getFirst().getMeasurementList().getNames()) {
+            outputList.put(name,
+                    pathObjects.stream()
+                            .mapToDouble(po ->
+                                    po.getMeasurementList().get(name) * weights.get(po)
+                            )
+                            .sum()
+            );
+        }
+    });
+
+    private static Map<PathObject, Double> computeAreaWeights(List<? extends PathObject> pathObjects) {
+        Logger logger = LoggerFactory.getLogger(MeasurementStrategy.class);
+        Map<PathObject, Double> output = new HashMap<>();
+        double sum = 0;
+        for (var po: pathObjects) {
+            double area = po.getROI().getArea();
+            sum += area;
+            output.put(po, area);
+        }
+        if (sum == 0) {
+            logger.warn("Zero areas");
+        }
+        for (var key: output.keySet()) {
+            double finalSum = sum;
+            output.computeIfPresent(key, (k, value) -> value / finalSum);
+        }
+        return output;
+    }
+
+    /**
+     * Calculate the median of each measurement and pass these to the output object.
+     */
+    MeasurementStrategy MEDIAN = ((pathObjects, outputList) -> {
+        for (var name: pathObjects.getFirst().getMeasurementList().getNames()) {
+            outputList.put(
+                    name,
+                    pathObjects.stream()
+                            .mapToDouble(po -> po.getMeasurementList().get(name))
+                            .sorted()
+                            .skip((pathObjects.size() - 1) / 2)
+                            .limit(2 - pathObjects.size() % 2)
+                            .average()
+                            .orElse(Double.NaN)
+            );
+        }
+    });
+
+    /**
+     * Choose an object at random and use the measurements from this object.
+     */
+    MeasurementStrategy RANDOM = ((pathObjects, outputList) -> {
```

Comment:
Is there ever a time when this is desirable? Can keep it, but I'm curious... I suppose `USE_FIRST` *could* be essentially random., depending upon how we figure out the input